app/backend/src/couchers/models.py

```diff
@@ -654,3 +654,324 @@ class InitiatedUpload(Base):
     @hybrid_property
     def is_valid(self):
         return (self.created <= func.now()) & (self.expiry >= func.now())
+
+
+class Node(Base):
+    """
+    Node, i.e. geographical subdivision of the world
+    """
+
+    __tablename__ = "nodes"
+
+    id = Column(BigInteger, primary_key=True)
+
+    name = Column(String, nullable=False)
+    parent_node_id = Column(ForeignKey("nodes.id"), nullable=True, index=True)
+    geom = Column(Geometry(geometry_type="POLYGON", srid=4326), nullable=False)
+    official_cluster_id = Column(ForeignKey("clusters.id"), nullable=False, unique=True, index=True)
+    created = Column(DateTime(timezone=True), nullable=False, server_default=func.now())
+
+    parent_node = relationship("Node", backref="child_nodes", remote_side="Node.id")
+    official_cluster = relationship("Cluster", backref="node", uselist=False)
+
+
+class Cluster(Base):
+    """
+    Cluster, administered grouping of content
+    """
+
+    __tablename__ = "clusters"
+
+    id = Column(BigInteger, primary_key=True)
+    name = Column(String, nullable=False)
+    main_page_id = Column(ForeignKey("pages.id"), nullable=False, unique=True, index=True)
+    created = Column(DateTime(timezone=True), nullable=False, server_default=func.now())
+
+    main_page = relationship("Page", backref="cluster", uselist=False, foreign_keys="Cluster.main_page_id")
+
+    nodes = relationship("Cluster", backref="clusters", secondary="node_cluster_associations")
+    pages = relationship("Page", backref="clusters", secondary="cluster_page_associations")
+    events = relationship("Event", backref="clusters", secondary="cluster_event_associations")
+    discussions = relationship("Discussion", backref="clusters", secondary="cluster_discussion_associations")
+    subscribers = relationship("User", backref="clusters", secondary="cluster_subscriptions")
+
+
+class NodeClusterAssociation(Base):
+    """
+    NodeClusterAssociation, grouping of nodes
+    """
+
+    __tablename__ = "node_cluster_associations"
+    __table_args__ = (UniqueConstraint("node_id", "cluster_id"),)
+
+    id = Column(BigInteger, primary_key=True)
+
+    node_id = Column(ForeignKey("nodes.id"), nullable=False, index=True)
+    cluster_id = Column(ForeignKey("clusters.id"), nullable=False, index=True)
+
+    node = relationship("Node", backref="node_cluster_associations")
+    cluster = relationship("Cluster", backref="node_cluster_associations")
```

Comment:
Do we need these in the `*Association` tables if we're using the `secondary` relationships above?

Reply:
No, those three tables will probably not need it. But I think it is best to do. Then the rule is simple: every foreign key gets a backref relationship. Some other such relationship might also not be needed and probably not in both directions.